docs/RFCS/schema_gossip.md

```diff
@@ -0,0 +1,40 @@
+- Feature Name: schema_distribution
+- Status: unstarted
+- Start Date: TBD
+- RFC PR: #1743
+- Cockroach Issue:
+
+# Summary
+
+This RFC suggests implementing eventually-consistent replication of the SQL schema to all Cockroach nodes in a cluster using gossip.
+
+# Motivation
+
+In order to support performant SQL queries, each gateway node must be able to address the data requested by the query. Today this requires the node to read `TableDescriptor`s from the KV map, which we believe (though we haven't measured) will cause a substantial performance hit which we can mitigate by actively gossiping the necessary metadata.
+
+# Detailed design
+
+The entire `keys.SystemDBSpan` span will have its writes bifurcated into gossip (with a TTL TBD) in the same way that `storage.(*Replica).maybeGossipConfigs()` works today. The gossip system will provide atomicity in propagating these modifications through the usual sequence number mechanism.
+
+Complete propagation of new metadata will take at most `numHops * gossipInterval` where `numHops` is the maximum number of hops between any node and the publishing node, and `gossipInterval` is the maximum interval between sequential writes to the gossip network on a given node.
+
+On the read side, metadata reads' behaviour will change such that they will read from gossip rather than the KV store. This will require plumbing a closure or a reference to the `Gossip` instance down to the `sql.Planner` instance.
```

Comment:
Sure. I'm okay with omitting that detail from this RFC since there's no controversy; either metadata-direct operations always do consistent things or we add flags.